src/ir_Technibel.cpp

```diff
@@ -0,0 +1,416 @@
+// Copyright 2020 Quentin Briollant
+
+/// @file
+/// @brief Support for Technibel protocol.
+
+#include "ir_Technibel.h"
+#include "IRrecv.h"
+#include "IRsend.h"
+#include "IRtext.h"
+#include "IRutils.h"
+#include <algorithm>
+
+using irutils::addBoolToString;
+using irutils::addModeToString;
+using irutils::addFanToString;
+using irutils::addLabeledString;
+using irutils::addTempToString;
+using irutils::minsToString;
+using irutils::setBit;
+using irutils::setBits;
+
+const uint16_t kTechnibelAcHdrMark = 8836;;
+const uint16_t kTechnibelAcHdrSpace = 4380;;
+const uint16_t kTechnibelAcBitMark = 523;;
+const uint16_t kTechnibelAcOneSpace = 1696;;
+const uint16_t kTechnibelAcZeroSpace = 564;;
+const uint32_t kTechnibelAcGap = kDefaultMessageGap;
+const uint16_t kTechnibelAcFreq = 38000;
+const uint16_t kTechnibelAcOverhead = 3;
+
+
+#if SEND_TECHNIBEL_AC
+// Send an Technibel AC formatted message.
+//
+// Args:
+//   data:   The message to be sent.
+//   nbits:  The number of bits of the message to be sent.
+//           Typically kTechnibelAcBits.
+//   repeat: The number of times the command is to be repeated.
+//
+// Status: STABLE / Reported as working on a real device.
+void IRsend::sendTechnibelAc(const uint64_t data, const uint16_t nbits,
+                            const uint16_t repeat) {
+  sendGeneric(kTechnibelAcHdrMark, kTechnibelAcHdrSpace,
+              kTechnibelAcBitMark, kTechnibelAcOneSpace,
+              kTechnibelAcBitMark, kTechnibelAcZeroSpace,
+              kTechnibelAcBitMark, kTechnibelAcGap,
+              data, nbits, kTechnibelAcFreq, false,  // LSB First.
```

Comment:
As mentioned elsewhere, using MSB First will negate having to use `reverseBits()` elsewhere.
e.g.
`              data, nbits, kTechnibelAcFreq, true,  // MSB First.`